### ISSUE TITLE
Store other genotype fields on storage-mongodb

### ISSUE THREAD

**Issue by j-coll** [opencb/opencga#221]
The storage-mongodb was designed to store, at the beginning, all the samples information available on the VCF. For performance reasons, a really big compression was made by storing just the genotypes in arrays of sampleIds that contains one common genotype. With this we could save the same number of variants (with just GT, like the 1000g files) in 7x less space.

But, there was not possible to store other fields any more.
MongoDB technology has improved, and there are some cases that store this information is mandatory.

In order to do this, few small changes on the variant MongoDB schema are required.
The information will be stored in positional arrays, where each position contains the value of this field for a sample. Due to the variability of this data (in comparison with the genotypes that use to be less than 4 values), it's not profitable to do the same trick made with the genotypes, where the value is stored as the key.

``` json
{
  "_id" : "22_1234_A_T",
  "studies" : [ 
    { 
      "sid" : "1000g",  
      "files" : [ {"fid" : 23 }, {"fid" : 24 }],
      "gt": {"0|1" : [3, 5], "0|0" : [2, 4] },
      "dp" : [13, 21, 14, 7] 
    } 
  ]
}
```

We must be aware that samples can come from multiple files, so is needed to ensure the loading order of the files, and the samples position in files (which is not needed to be the same as in the original file). The position and the sampleId it not needed to be the same. For the example above, sample ids are `[2, 3]` from file 23, and `[4, 5]` from file 24, and positions are from 0 to 3. 

Also, is needed to represent missing values, to ensure the sample positions in case that a variant is not present on a specific variant. A missing value can be represented with a `-1`, a `null` or a `"."`.

``` json
{
  "_id" : "22_1234_A_T",
  "studies" : [ 
    { 
      "sid" : "1000g",  
      "files" : [ {"fid" : 23 } ],
      "gt": {"1|1" : [2, 3], "?/?" : [4, 5] },
      "dp" : [13, 21, -1, -1] 
    } 
  ]
}
```

To summarize the list of actions:
 * Ensure samples position and file loading order for each study
 * Accept and save a list of "*extra fields*" to be stored for each study.
 * Fill positional arrays with missing values if needed.
 * Change the SampleData converter to include the requested information.

**Comment by j-coll** [comment]
A new constraint has been added to Storage-MongoDB:

MongoDB storage plugin is not able to load batches of samples in a unordered way.
A batch of samples is a group of samples of any size. It may be composed of one or several VCF files, depending on whether it is split by region (horizontally) or not.
All the files from the same batch must be loaded before loading the next batch. If a new batch of samples begins to be loaded, it won't be possible to load other files from previous batches.

**Comment by j-coll** [comment]
While indexing the first file of an study (so, when creating a new study in opencta-storage) a new param can be specified: `--include-extra-fields [csv]` , which can define a list con other fields that should be loaded.
These will be the same extra fields stored for all the next indexed files. No fields can be added or deleted after the first file is indexed.
Because this information is saved as metadata (in the StudyConfiguration) it is not needed for the next indexed files.
